data-prepper-plugins/kinesis-source/src/integrationTest/java/org/opensearch/dataprepper/plugins/kinesis/source/KinesisSourceIT.java

```diff
@@ -0,0 +1,286 @@
+
+        when(kinesisStreamConfig.getName()).thenReturn(streamName);
+        when(kinesisStreamConfig.getCheckPointInterval()).thenReturn(CHECKPOINT_INTERVAL);
+        when(kinesisStreamConfig.getInitialPosition()).thenReturn(InitialPositionInStream.TRIM_HORIZON);
+        when(kinesisSourceConfig.getConsumerStrategy()).thenReturn(ConsumerStrategy.ENHANCED_FAN_OUT);
+        when(kinesisSourceConfig.getStreams()).thenReturn(List.of(kinesisStreamConfig));
+        buffer = mock(Buffer.class);
+        kinesisLeaseConfigSupplier = mock(KinesisLeaseConfigSupplier.class);
+        kinesisLeaseConfig = mock(KinesisLeaseConfig.class);
+        workerIdentifierGenerator = mock(WorkerIdentifierGenerator.class);
+        kinesisLeaseCoordinationTableConfig = mock(KinesisLeaseCoordinationTableConfig.class);
+        when(kinesisLeaseConfig.getLeaseCoordinationTable()).thenReturn(kinesisLeaseCoordinationTableConfig);
+        when(kinesisLeaseCoordinationTableConfig.getTableName()).thenReturn(leaseTableName);
+        when(kinesisLeaseCoordinationTableConfig.getRegion()).thenReturn(System.getProperty(AWS_REGION));
+        when(kinesisLeaseCoordinationTableConfig.getAwsRegion()).thenReturn(Region.of(System.getProperty(AWS_REGION)));
+        when(kinesisLeaseConfigSupplier.getKinesisExtensionLeaseConfig()).thenReturn(Optional.ofNullable(kinesisLeaseConfig));
+
+        codec = mock(InputCodec.class);
+        PluginModel pluginModel = mock(PluginModel.class);
+        when(pluginModel.getPluginName()).thenReturn(codec_plugin_name);
+        when(pluginModel.getPluginSettings()).thenReturn(Collections.emptyMap());
+        when(kinesisSourceConfig.getCodec()).thenReturn(pluginModel);
+        when(kinesisSourceConfig.isAcknowledgments()).thenReturn(false);
+        when(kinesisSourceConfig.getNumberOfRecordsToAccumulate()).thenReturn(NUMBER_OF_RECORDS_TO_ACCUMULATE);
+        when(kinesisSourceConfig.getBufferTimeout()).thenReturn(Duration.ofMillis(1));
+
+        kinesisClientFactory = mock(KinesisClientFactory.class);
+        when(kinesisClientFactory.buildDynamoDBClient(kinesisLeaseCoordinationTableConfig.getAwsRegion())).thenReturn(DynamoDbAsyncClient.builder()
+                .region(Region.of(System.getProperty(AWS_REGION)))
+                .build());
+        when(kinesisClientFactory.buildKinesisAsyncClient(awsAuthenticationConfig.getAwsRegion())).thenReturn(KinesisClientUtil.createKinesisAsyncClient(
+                KinesisAsyncClient.builder()
+                        .region(Region.of(System.getProperty(AWS_REGION)))
+        ));
+        when(kinesisClientFactory.buildCloudWatchAsyncClient(kinesisLeaseCoordinationTableConfig.getAwsRegion())).thenReturn(CloudWatchAsyncClient.builder()
+                .region(Region.of(System.getProperty(AWS_REGION)))
+                .build());
+
+        when(pipelineDescription.getPipelineName()).thenReturn(PIPELINE_NAME);
+
+        String workerId = "worker-identifier-" + testId;
```

Comment:
nit: final

Reply:
Thanks @chenqi0805! I will address this.